### PR TITLE
EA Tag hover over redesign

### PR DESCRIPTION
**Pull request by oetherington**
This PR adds the redesigned tag hover over, along with quite a lot of refactoring. [Relevant Figma](https://www.figma.com/file/ZqlWFgIlVJZhpqEL4v9f1A/Q3.2-2023?node-id=344%3A21511&mode=dev).

With relevance:
Screenshot 2023-10-25 at 20 26 24

Without relevance:
Screenshot 2023-10-25 at 20 26 28

Without a description:
Screenshot 2023-10-25 at 20 26 35

Frontpage/personal dummy tags:
Screenshot 2023-10-25 at 20 28 20

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205804821390931) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/tagging/TagTooltipWrapper.tsx
Comment (oetherington) [inline packages/lesswrong/components/tagging/TagTooltipWrapper.tsx:1]: Replaced with `TagsTooltip`

**Comment by oetherington** [comment]
@agnesstenlund This should be ready now.

The designs specified 7 lines for the preview, but we can't really cut it off based on the number of lines (like with the post tooltip) so we're using characters instead that seems to be _about_ 7 lines.

**Review by s-cheng — CHANGES_REQUESTED** [review]
Looks very nice on the EA Forum side! (Though I hope we add preview images at some point ^^)

LW looks quite broken - I'd like to test LW more thoroughly, so ask me for another review once you've fixed that side.
Screen Shot 2023-11-07 at 12 15 46 PM

**Comment by agnesstenlund** [comment]
Looks perfect <3

**Comment by oetherington** [comment]
@s-cheng I think everything here should be fixed now.

CC @agnesstenlund in case you want to take another look (but nothing should have changed for EA forum since you last looked except bug fixes).

@b0b3rt We created this PR a couple of months ago to redesign the tags tooltip on the EA forum, but in the process we broke the tooltip pretty badly for LW and fixing it got deprioritised in favour of other stuff. I think I've now fixed the problems on LW, but I'd appreciate if you could just give it a quick look in case I missed something (not super urgent though). Thanks!

**Comment by s-cheng** [comment]
@oetherington Thanks! It's def less broken but I still see some discrepancies and I think it's important for us not to mess with LW's site. I noticed three small things:
1. Looks like there is additional padding in the LW tooltip now
2. Looks like the LW tooltip is aligned with the bottom of the tag name rather than the bottom of the tag pill now
3. Possibly the tag description is not getting truncated early enough? I'm not sure

Compare the version on LW prod:
Screen Shot 2024-01-11 at 3 19 38 PM

With this branch:
Screen Shot 2024-01-11 at 3 20 00 PM

**Comment by oetherington** [comment]
@s-cheng Whoops - missed those! 1 and 2 are now fixed. 3 is the same as master.

AFAIK, the only difference now is that post titles/authors that are too long are now cutoff at the edge of the padding instead of extending all the way to the edge of the tooltip. I think the previous behaviour was actually just a bug and that it looks better this way, but I'll happily change it back if anybody from LW objects.

**Review by s-cheng — APPROVED** [review]
🙌

**Pull request merged**